doc/make.jl

```diff
@@ -46,18 +46,21 @@ end
 const render_pdf = "pdf" in ARGS
 
 # Generate a suitable markdown file from NEWS.md and put it in src
-str = read(joinpath(@__DIR__, "..", "NEWS.md"), String)
-splitted = split(str, "<!--- generated by NEWS-update.jl: -->")
-@assert length(splitted) == 2
-replaced_links = replace(splitted[1], r"\[\#([0-9]*?)\]" => s"[#\g<1>](https://github.com/JuliaLang/julia/issues/\g<1>)")
-write(
-    joinpath(@__DIR__, "src", "NEWS.md"),
-    """
-    ```@meta
-    EditURL = "https://github.com/JuliaLang/julia/blob/master/NEWS.md"
-    ```
-
-    """ * replaced_links)
+function generate_markdown(basename)
+    str = read(joinpath(@__DIR__, "..", "$basename.md"), String)
+    splitted = split(str, "<!--- generated by $basename-update.jl: -->")
+    @assert length(splitted) == 2
+    replaced_links = replace(splitted[1], r"\[\#([0-9]*?)\]" => s"[#\g<1>](https://github.com/JuliaLang/julia/issues/\g<1>)")
+    write(
+        joinpath(@__DIR__, "src", "$basename.md"),
+        """
+        ```@meta
+        EditURL = "https://github.com/JuliaLang/julia/blob/master/$basename.md"
+        ```
+    
```

Comment:
```suggestion

```

Reply:
Why should the empty line be removed?

My PR doesn't change the output, only how it is generated. Shouldn't a change to the generated output be a different PR?

Or am I misinterpreting your suggestion? Do you want to remove trailing white space only, keeping the line?

Reply:
> Do you want to remove trailing white space only, keeping the line?

Yes, and my local tests show that these multi-line strings are also equivalent after deleting these spaces.

```julia
s0 =
    """
    ```@meta
    EditURL = "https://github.com/JuliaLang/julia/blob/master/NEWS.md"
    ```

    """
s1 =
        """
        ```@meta
        EditURL = "https://github.com/JuliaLang/julia/blob/master/NEWS.md"
        ```
    
        """
s2 =
        """
        ```@meta
        EditURL = "https://github.com/JuliaLang/julia/blob/master/NEWS.md"
        ```

        """
[s0, s1, s2]
# 3-element Vector{String}:
#  "```@meta\nEditURL = \"https://github.com/JuliaLang/julia/blob/master/NEWS.md\"\n```\n\n"
#  "```@meta\nEditURL = \"https://github.com/JuliaLang/julia/blob/master/NEWS.md\"\n```\n    \n"
#  "```@meta\nEditURL = \"https://github.com/JuliaLang/julia/blob/master/NEWS.md\"\n```\n\n"

s00 = 
    """
    ```@meta
    url=""
    ```

    """

s01 = 
    """
    ```@meta
    url=""
    ```
    
    """

s10 = 
        """
        ```@meta
        url=""
        ```

        """

s11 = 
        """
        ```@meta
        url=""
        ```
    
        """

s12 = 
        """
        ```@meta
        url=""
        ```
        
        """

[s00, s01, s10, s11, s12]
# 5-element Vector{String}:
#  "```@meta\nurl=\"\"\n```\n\n"
#  "```@meta\nurl=\"\"\n```\n\n"
#  "```@meta\nurl=\"\"\n```\n\n"
#  "```@meta\nurl=\"\"\n```\n    \n"
#  "```@meta\nurl=\"\"\n```\n\n"
```